internal/infrastructure/runner/runner.go

```diff
@@ -45,8 +45,10 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 		r.Logger.Info("received a notification")
 		in := r.InfraIR.Get()
 		if in == nil {
-			r.Logger.Info("ir is nil, skipping")
-			continue
```

Comment:
For the empty infra case, I think the IR producer should generate a key with nil value, rather than setting an empty map
This current implementation is going to result in this code trying to delete the infra during init/start which will fail because the deployment doesn't exist